rasa/shared/core/training_data/structures.py

```diff
@@ -86,15 +86,13 @@ def __init__(
         end_checkpoints: Optional[List[Checkpoint]] = None,
         events: Optional[List[Union[Event, List[Event]]]] = None,
         source_name: Optional[Text] = None,
-        is_rule: bool = None,
     ) -> None:
 
         self.end_checkpoints = end_checkpoints if end_checkpoints else []
         self.start_checkpoints = start_checkpoints if start_checkpoints else []
         self.events = events if events else []
         self.block_name = block_name
         self.source_name = source_name
-        self.is_rule = is_rule
```

Comment:
Do you think we could leave this variable as-is? It's useful for checking if we're dealing with rules or not, without having to use `isinstance`.

Reply:
I wanted to separate the `Rules` from non-`Rules` completely, just to clearly abstract all rule-related stuff.
The `get_rules_condition` and `get_rules_steps` inside the `StoryStep` class are my mistakes, I'll remove them 😬